packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -0,0 +1,188 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../base/logger.dart';
+import '../base/os.dart';
+import '../base/process.dart';
+import '../base/utils.dart';
+import '../build_info.dart';
+import '../cache.dart';
+import '../globals.dart';
+import 'android_sdk.dart';
+
+const String gradleManifestPath = 'android/app/src/main/AndroidManifest.xml';
+const String gradleAppOut = 'android/app/build/outputs/apk/app-debug.apk';
+
+bool isProjectUsingGradle() {
+  return FileSystemEntity.isFileSync('android/build.gradle');
+}
+
+String locateSystemGradle({ bool ensureExecutable: true }) {
+  String gradle = _locateSystemGradle();
+  if (ensureExecutable && gradle != null) {
+    File file = new File(gradle);
+    if (file.existsSync())
+      os.makeExecutable(file);
```

Comment:
This is one of the issues I ran into (https://github.com/flutter/flutter/issues/4429) when first trying on the gradle flow - the gradle script on my machine wasn't executable.

It's a bash script, so instead of executing it, we could run it in bash. Or we could fail when it tries to run (w/o the executable bit), and provide a good message to the user and steps to fix.